src/main/java/com/volasoftware/tinder/service/FriendServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.volasoftware.tinder.service;
+
+import com.volasoftware.tinder.model.User;
+import com.volasoftware.tinder.model.UserType;
+import com.volasoftware.tinder.repository.UserRepository;
+import com.volasoftware.tinder.utility.BotGenerator;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class FriendServiceImpl implements FriendService {
+
+    private final UserService userService;
+    private final UserRepository userRepository;
+
+    public FriendServiceImpl(UserService userService, UserRepository userRepository) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+    }
+
+    public void seedFriend(@RequestParam(required = false) Long userId) {
```

Comment:
Can you check the userId first and put the whole logic after it. Than way the code will no do extra stuff before recognize the missing user for example